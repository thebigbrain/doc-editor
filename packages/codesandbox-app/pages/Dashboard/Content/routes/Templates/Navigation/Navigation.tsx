import React from 'react';
import { Container, NavigationLink, Number } from './elements';

interface INavigationProps {
  teamId?: string;
  number?: number;
}

export const Navigation = ({ teamId, number }: INavigationProps) => (
  <Container>
    <NavigationLink
      to={
        teamId ? `/dashboard/teams/${teamId}/templates` : `/dashboard/templates`
      }
    >
      {teamId ? 'Team Templates' : 'My Templates'}
    </NavigationLink>

    {number == null && <Number>{number}</Number>}
  </Container>
);
