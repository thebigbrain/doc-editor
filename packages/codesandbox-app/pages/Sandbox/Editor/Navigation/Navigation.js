import React from 'react'
import {GoPlus as PlusIcon} from 'react-icons/go'
import Tooltip from '@codesandbox/common/lib/components/Tooltip'
// import getWorkspaceItems, { getDisabledItems } from '~/store/modules/workspace/items';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import InfoIcon from '@codesandbox/common/lib/icons/sandbox.svg'
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import GitHubIcon from '@codesandbox/common/lib/icons/github.svg'
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import LiveIcon from '@codesandbox/common/lib/icons/live.svg'
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import FilesIcon from '@codesandbox/common/lib/icons/file.svg'
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import RocketIcon from '@codesandbox/common/lib/icons/rocket.svg'
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import ConfigurationIcon from '@codesandbox/common/lib/icons/cog.svg'
import {Container, IconContainer, Separator} from './elements'
import ServerIcon from './ServerIcon'
import {useOvermind} from "~/overmind"

const IDS_TO_ICONS = {
  project: InfoIcon,
  'project-summary': InfoIcon,
  files: FilesIcon,
  github: GitHubIcon,
  deploy: RocketIcon,
  config: ConfigurationIcon,
  live: LiveIcon,
  more: PlusIcon,
  server: ServerIcon,
}

const IconComponent =
  ({
     item,
     isDisabled,
   }) => {
    const {
      state: store,
      actions: {
        workspace: {setWorkspaceHidden, setWorkspaceItem},
      }
    } = useOvermind()
    const {id, name} = item

    const Icon = IDS_TO_ICONS[id]
    const selected =
      !store.workspace.workspaceHidden &&
      id === store.workspace.openedWorkspaceItem
    return (
      <Tooltip key={id} placement="right" content={name}>
        <IconContainer
          isDisabled={isDisabled}
          selected={selected}
          onClick={() => {
            if (selected) {
              setWorkspaceHidden({hidden: true})
            } else {
              setWorkspaceHidden({hidden: false})
              setWorkspaceItem({item: id})
            }
          }}
        >
          <Icon/>
        </IconContainer>
      </Tooltip>
    )
  }

export const Navigation =
  ({
     topOffset,
     bottomOffset,
   }) => {
    const {state: store, actions} = useOvermind()

    const shownItems = actions.editor.getWorkspaceItems()
    const disabledItems = actions.editor.getDisabledItems()

    return (
      <Container topOffset={topOffset} bottomOffset={bottomOffset}>
        {shownItems.map(item => (
          <IconComponent key={item.id} item={item}/>
        ))}

        {disabledItems.length > 0 && <Separator/>}

        {disabledItems.map(item => (
          <IconComponent key={item.id} item={item} isDisabled/>
        ))}
      </Container>
    )
  }
