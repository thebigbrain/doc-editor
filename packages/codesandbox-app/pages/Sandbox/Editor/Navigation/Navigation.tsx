import React from 'react';
import { hooksObserver, inject } from 'app/componentConnectors';
import PlusIcon from 'react-icons/lib/go/plus';
import Tooltip from '@codesandbox/common/lib/components/Tooltip';
import getWorkspaceItems, { getDisabledItems, INavigationItem } from 'app/store/modules/workspace/items';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import InfoIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/sandbox.svg';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import GitHubIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/github.svg';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import LiveIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/live.svg';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import FilesIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/file.svg';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import RocketIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/rocket.svg';
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import ConfigurationIcon from '-!svg-react-loader!@codesandbox/common/lib/icons/cog.svg';
import { Container, IconContainer, Separator } from './elements';
import ServerIcon from './ServerIcon';

const IDS_TO_ICONS = {
  project: InfoIcon,
  'project-summary': InfoIcon,
  files: FilesIcon,
  github: GitHubIcon,
  deploy: RocketIcon,
  config: ConfigurationIcon,
  live: LiveIcon,
  more: PlusIcon,
  server: ServerIcon,
};

interface IconProps {
  item: INavigationItem;
  isDisabled?: boolean;
  store: any;
  signals: any;
}

const IconComponent = inject('store', 'signals')(
  hooksObserver(
    ({
       item,
       isDisabled,
       store,
       signals: {
         workspace: { setWorkspaceHidden, setWorkspaceItem },
       },
     }: IconProps) => {
      const { id, name } = item;

      const Icon = IDS_TO_ICONS[id];
      const selected =
        !store.workspace.workspaceHidden &&
        id === store.workspace.openedWorkspaceItem;
      return (
        <Tooltip key={id} placement="right" content={name}>
          <IconContainer
            isDisabled={isDisabled}
            selected={selected}
            onClick={() => {
              if (selected) {
                setWorkspaceHidden({ hidden: true });
              } else {
                setWorkspaceHidden({ hidden: false });
                setWorkspaceItem({ item: id });
              }
            }}
          >
            <Icon/>
          </IconContainer>
        </Tooltip>
      );
    },
  ),
);

export const Navigation = inject('store')(
  hooksObserver(
    ({
       topOffset,
       bottomOffset,
       store,
     }: {
      topOffset: number;
      bottomOffset: number;
      store: any;
    }) => {
      const shownItems = getWorkspaceItems(store);
      const disabledItems = getDisabledItems(store);

      return (
        <Container topOffset={topOffset} bottomOffset={bottomOffset}>
          {shownItems.map(item => (
            <IconComponent key={item.id} item={item}/>
          ))}

          {disabledItems.length > 0 && <Separator/>}

          {disabledItems.map(item => (
            <IconComponent key={item.id} item={item} isDisabled/>
          ))}
        </Container>
      );
    },
  ),
);
