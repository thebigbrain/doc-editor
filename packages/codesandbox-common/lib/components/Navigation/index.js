"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Logo = _interopRequireDefault(require("../Logo"));

var _MaxWidth = _interopRequireDefault(require("../flex/MaxWidth"));

var _media = _interopRequireDefault(require("../../utils/media"));

var _analytics = _interopRequireDefault(require("../../utils/analytics"));

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 1.75em;\n  height: 1.75em;\n  border-radius: 4px;\n  margin-left: 0.75rem;\n  margin-bottom: 0;\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      @media (max-width: ", "px) {\n        display: none;\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      display: none;\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    font-size: 1rem;\n    margin: 0 .5rem;\n  "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      transition: 0.3s ease all;\n      padding: 0.2rem 0.8rem;\n      border-radius: 4px;\n      font-weight: 600;\n      background-color: ", ";\n      border: 2px solid rgba(255, 255, 255, 0.3);\n\n      &:hover {\n        color: white;\n        background-color: #7fc3f7;\n        border-color: transparent;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  align-items: center;\n  transition: 0.2s ease color;\n  font-size: 1.125rem;\n  text-decoration: none;\n  color: white;\n\n  margin: 0 1rem;\n  font-weight: 400;\n\n  &:hover {\n    color: ", ";\n  }\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    width: 38px;\n    height: 38px;\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  color: white;\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n  flex: auto;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n  padding: 1rem 0;\n  width: 100%;\n  color: white;\n  z-index: 5;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Container = _styledComponents["default"].div(_templateObject());

var Left = _styledComponents["default"].div(_templateObject2());

var StyledLogo = (0, _styledComponents["default"])(_Logo["default"])(_templateObject3(), _media["default"].phone(_templateObject4()));

var Item = _styledComponents["default"].a(_templateObject5(), function (props) {
  return props.theme.secondary;
}, function (props) {
  return props.button && (0, _styledComponents.css)(_templateObject6(), props.theme.secondary);
}, _media["default"].phone(_templateObject7()), function (props) {
  return props.hidePhone && (0, _styledComponents.css)(_templateObject8(), _media["default"].phone(_templateObject9()));
}, function (props) {
  return props.hideOn && (0, _styledComponents.css)(_templateObject10(), props.hideOn);
});

var Right = _styledComponents["default"].div(_templateObject11());

var Image = _styledComponents["default"].img(_templateObject12());

var Navigation =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2["default"])(Navigation, _React$PureComponent);

  function Navigation() {
    var _this;

    (0, _classCallCheck2["default"])(this, Navigation);
    _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Navigation).apply(this, arguments));
    _this.state = {
      user: null
    };

    _this.fetchCurrentUser = function () {
      var jwt = JSON.parse(localStorage.getItem('jwt'));
      var BASE = process.env.NODE_ENV === 'development' ? 'http://localhost:3000' : '';
      window.fetch(BASE + '/api/v1/users/current', {
        headers: {
          Authorization: "Bearer ".concat(jwt)
        }
      }).then(function (x) {
        return x.json();
      }).then(function (_ref) {
        var data = _ref.data;
        return _this.setState({
          user: data
        });
      })["catch"](function () {
        /* do nothing */
      });
    };

    return _this;
  }

  (0, _createClass2["default"])(Navigation, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (localStorage.getItem('jwt')) {
        this.fetchCurrentUser();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var user = this.state.user;
      return _react["default"].createElement(_MaxWidth["default"], {
        width: 1440
      }, _react["default"].createElement(Container, null, _react["default"].createElement(Left, null, _react["default"].createElement("a", {
        href: "/"
      }, _react["default"].createElement(StyledLogo, {
        width: 40,
        height: 40,
        style: {
          marginRight: '1rem'
        }
      })), _react["default"].createElement(Item, {
        href: "/explore"
      }, "Explore"), _react["default"].createElement(Item, {
        href: "/search"
      }, "Search"), _react["default"].createElement(Item, {
        href: "/docs"
      }, "Docs"), _react["default"].createElement(Item, {
        href: "/blog"
      }, "Blog"), _react["default"].createElement(Item, {
        href: "https://github.com/codesandbox/codesandbox-client",
        target: "_blank",
        rel: "noopener noreferrer",
        hideOn: 970
      }, "GitHub"), _react["default"].createElement(Item, {
        href: "/jobs"
      }, "Careers")), _react["default"].createElement(Right, null, !user && _react["default"].createElement(Item, {
        hideOn: 875,
        href: "/signin"
      }, "Sign In"), _react["default"].createElement(Item, {
        onClick: function onClick() {
          (0, _analytics["default"])('Navigation - Create Sandbox Clicked');
        },
        hidePhone: true,
        href: "/s",
        rel: "noopener noreferrer",
        button: !user
      }, "Create Sandbox"), user && _react["default"].createElement(Item, {
        hidePhone: true,
        href: "/dashboard",
        rel: "noopener noreferrer"
      }, user.username, _react["default"].createElement(Image, {
        alt: user.username,
        src: user.avatar_url
      })))));
    }
  }]);
  return Navigation;
}(_react["default"].PureComponent);

exports["default"] = Navigation;