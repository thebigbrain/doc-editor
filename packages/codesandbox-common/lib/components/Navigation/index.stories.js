"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _ = _interopRequireDefault(require("."));

var stories = (0, _react2.storiesOf)('components/Navigation', module);
stories.add('Navigation', function () {
  return _react["default"].createElement(_["default"], null);
});