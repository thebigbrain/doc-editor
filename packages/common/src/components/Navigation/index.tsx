import React from 'react';
import styled, { css } from 'styled-components';

import Logo from '../Logo';
import MaxWidth from '../flex/MaxWidth';

import media from '../../utils/media';
import track from '../../utils/analytics';

const Container = styled.div`
  display: flex;
  align-items: center;
  padding: 1rem 0;
  width: 100%;
  color: white;
  z-index: 5;
`;

const Left = styled.div`
  display: flex;
  align-items: center;
  flex: auto;
`;

const StyledLogo = styled(Logo)`
  color: white;
  ${media.phone`
    width: 38px;
    height: 38px;
  `};
`;

const Item = styled.a<{
  button?: boolean;
  hidePhone?: boolean;
  hideOn?: number;
}>`
  display: inline-flex;
  align-items: center;
  transition: 0.2s ease color;
  font-size: 1.125rem;
  text-decoration: none;
  color: white;

  margin: 0 1rem;
  font-weight: 400;

  &:hover {
    color: ${props => props.theme.secondary};
  }

  ${props =>
  props.button &&
  css`
      transition: 0.3s ease all;
      padding: 0.2rem 0.8rem;
      border-radius: 4px;
      font-weight: 600;
      background-color: ${props.theme.secondary};
      border: 2px solid rgba(255, 255, 255, 0.3);

      &:hover {
        color: white;
        background-color: #7fc3f7;
        border-color: transparent;
      }
    `};

  ${media.phone`
    font-size: 1rem;
    margin: 0 .5rem;
  `};

  ${props =>
  props.hidePhone &&
  css`
      ${media.phone`
      display: none;
    `};
    `};

  ${props =>
  props.hideOn &&
  css`
      @media (max-width: ${props.hideOn}px) {
        display: none;
      }
    `};
`;

const Right = styled.div`
  display: flex;
  align-items: center;
`;

const Image = styled.img`
  width: 1.75em;
  height: 1.75em;
  border-radius: 4px;
  margin-left: 0.75rem;
  margin-bottom: 0;
`;

export default class Navigation extends React.PureComponent {
  state = {
    user: null,
  };

  fetchCurrentUser = () => {
    const jwt = JSON.parse(localStorage.getItem('jwt'));

    const BASE =
      process.env.NODE_ENV === 'development' ? 'http://localhost:3000' : '';

    window
      .fetch(BASE + '/api/v1/users/current', {
        headers: { Authorization: `Bearer ${jwt}` },
      })
      .then(x => x.json())
      .then(({ data }) => this.setState({ user: data }))
      .catch(() => {
        /* do nothing */
      });
  };

  componentDidMount() {
    if (localStorage.getItem('jwt')) {
      this.fetchCurrentUser();
    }
  }

  render() {
    const { user } = this.state;
    return (
      <MaxWidth width={1440}>
        <Container>
          <Left>
            <a href="/">
              <StyledLogo
                width={40}
                height={40}
                style={{ marginRight: '1rem' }}
              />
            </a>

            <Item href="/explore">Explore</Item>
            <Item href="/search">Search</Item>
            <Item href="/docs">Docs</Item>

            <Item href="/blog">Blog</Item>

            <Item
              href="https://github.com/codesandbox/codesandbox-client"
              target="_blank"
              rel="noopener noreferrer"
              hideOn={970}
            >
              GitHub
            </Item>

            <Item href="/jobs">Careers</Item>
          </Left>

          <Right>
            {!user && (
              <Item hideOn={875} href="/signin">
                Sign In
              </Item>
            )}

            <Item
              onClick={() => {
                track('Navigation - Create Sandbox Clicked');
              }}
              hidePhone
              href="/s"
              rel="noopener noreferrer"
              button={!user}
            >
              Create Sandbox
            </Item>

            {user && (
              <Item hidePhone href="/dashboard" rel="noopener noreferrer">
                {user.username}
                <Image alt={user.username} src={user.avatar_url}/>
              </Item>
            )}
          </Right>
        </Container>
      </MaxWidth>
    );
  }
}
