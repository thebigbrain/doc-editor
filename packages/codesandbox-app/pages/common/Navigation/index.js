import * as React from 'react';
import { Link } from 'react-router-dom';
import Media from 'react-media';
import { exploreUrl, patronUrl, searchUrl } from '@codesandbox/common/lib/utils/url-generator';

import {
  GoSearch as SearchIcon,
  GoPlus as PlusIcon,
  GoFlame as FlameIcon,
} from 'react-icons/go';
import {
  MdNotifications as BellIcon
} from 'react-icons/md';
import Row from '@codesandbox/common/lib/components/flex/Row';
import Tooltip from '@codesandbox/common/lib/components/Tooltip';
import { HeaderSearchBar } from '~/components/HeaderSearchBar';
import { Overlay } from '~/components/Overlay';
// @ts-ignore
import PatronBadge from '@codesandbox/common/lib/utils/badges/svg/patron-4.svg';
import { Notifications } from './Notifications/index';

import { SignInButton } from '../SignInButton/index';
import { UserMenu } from '../UserMenu/index';
import { Action, Actions, Border, LogoWithBorder, Title, TitleWrapper, UnreadIcon, Wrapper } from './elements';
import { useOvermind } from '~/overmind'

export const Navigation = ({ title, searchNoInput, }) => {
  const {
    state: { isLoggedIn, isPatron, user, userNotifications },
    actions: { modalOpened, userNotifications: userNotificationsSignals }
  } = useOvermind()

  return (
    <Row justifyContent="space-between">
      <TitleWrapper>
        <a href="/?from-app=1">
          <LogoWithBorder height={35} width={35}/>
        </a>
        <Border/>
        <Title>{title}</Title>
      </TitleWrapper>
      <Wrapper>
        <Actions>
          <Action>
            <Media query="(max-width: 920px)">
              {matches =>
                matches || searchNoInput ? (
                  <Tooltip placement="bottom" content="Search All Sandboxes">
                    <Link style={{ color: 'white' }} to={searchUrl()}>
                      <SearchIcon height={35}/>
                    </Link>
                  </Tooltip>
                ) : (
                  <HeaderSearchBar/>
                )
              }
            </Media>
          </Action>

          <Action>
            <Tooltip placement="bottom" content="Explore Sandboxes">
              <a style={{ color: 'white' }} href={exploreUrl()}>
                <FlameIcon/>
              </a>
            </Tooltip>
          </Action>

          {!isPatron && (
            <Action>
              <Tooltip placement="bottom" content="Support CodeSandbox">
                <Link to={patronUrl()}>
                  <PatronBadge width={40} height={40}/>
                </Link>
              </Tooltip>
            </Action>
          )}

          {user && (
            <Overlay
              isOpen={userNotifications.notificationsOpened}
              content={Notifications}
              onOpen={userNotificationsSignals.notificationsOpened}
              onClose={userNotificationsSignals.notificationsClosed}
              event="Notifications"
              noHeightAnimation
            >
              {open => (
                <Observer>
                  {({ store }) => (
                    <Action
                      style={{ position: 'relative', fontSize: '1.25rem' }}
                      onClick={open}
                    >
                      <Tooltip
                        placement="bottom"
                        content={
                          store.userNotifications.unreadCount > 0
                            ? 'Show Notifications'
                            : 'No Notifications'
                        }
                      >
                        <BellIcon height={35}/>
                        {store.userNotifications.unreadCount > 0 && (
                          <UnreadIcon/>
                        )}
                      </Tooltip>
                    </Action>
                  )}
                </Observer>
              )}
            </Overlay>
          )}

          <Action
            style={{ fontSize: '1.125rem' }}
            onClick={() =>
              modalOpened({
                modal: 'newSandbox',
              })
            }
          >
            <Tooltip placement="bottom" content="New Sandbox">
              <PlusIcon height={35}/>
            </Tooltip>
          </Action>
        </Actions>

        {isLoggedIn ? <UserMenu/> : <SignInButton/>}
      </Wrapper>
    </Row>
  );
}
