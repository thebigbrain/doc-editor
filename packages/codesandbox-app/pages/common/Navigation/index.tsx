import * as React from 'react';
import { hooksObserver, inject, Observer } from 'app/componentConnectors';
import { Link } from 'react-router-dom';
import Media from 'react-media';
import { exploreUrl, patronUrl, searchUrl } from '@codesandbox/common/lib/utils/url-generator';

import SearchIcon from 'react-icons/lib/go/search';
import PlusIcon from 'react-icons/lib/go/plus';
import BellIcon from 'react-icons/lib/md/notifications';
import FlameIcon from 'react-icons/lib/go/flame';
import Row from '@codesandbox/common/lib/components/flex/Row';
import Tooltip from '@codesandbox/common/lib/components/Tooltip';
import { HeaderSearchBar } from 'app/components/HeaderSearchBar';
import { Overlay } from 'app/components/Overlay';
// @ts-ignore
import PatronBadge from '-!svg-react-loader!@codesandbox/common/lib/utils/badges/svg/patron-4.svg';
import { Notifications } from './Notifications/index';

import { SignInButton } from '../SignInButton/index';
import { UserMenu } from '../UserMenu/index';
import { Action, Actions, Border, LogoWithBorder, Title, TitleWrapper, UnreadIcon, Wrapper } from './elements';

interface Props {
  title: string;
  searchNoInput?: boolean;
  store: any;
  signals: any;
}

export const Navigation = inject('store', 'signals')(
  hooksObserver(
    ({
       title,
       searchNoInput,
       store: { isLoggedIn, isPatron, user, userNotifications },
       signals: { modalOpened, userNotifications: userNotificationsSignals },
     }: Props) => (
      <Row justifyContent="space-between">
        <TitleWrapper>
          <a href="/?from-app=1">
            <LogoWithBorder height={35} width={35}/>
          </a>
          <Border/>
          <Title>{title}</Title>
        </TitleWrapper>
        <Wrapper>
          <Actions>
            <Action>
              <Media query="(max-width: 920px)">
                {matches =>
                  matches || searchNoInput ? (
                    <Tooltip placement="bottom" content="Search All Sandboxes">
                      <Link style={{ color: 'white' }} to={searchUrl()}>
                        <SearchIcon height={35}/>
                      </Link>
                    </Tooltip>
                  ) : (
                    <HeaderSearchBar/>
                  )
                }
              </Media>
            </Action>

            <Action>
              <Tooltip placement="bottom" content="Explore Sandboxes">
                <a style={{ color: 'white' }} href={exploreUrl()}>
                  <FlameIcon/>
                </a>
              </Tooltip>
            </Action>

            {!isPatron && (
              <Action>
                <Tooltip placement="bottom" content="Support CodeSandbox">
                  <Link to={patronUrl()}>
                    <PatronBadge width={40} height={40}/>
                  </Link>
                </Tooltip>
              </Action>
            )}

            {user && (
              <Overlay
                isOpen={userNotifications.notificationsOpened}
                content={Notifications}
                onOpen={userNotificationsSignals.notificationsOpened}
                onClose={userNotificationsSignals.notificationsClosed}
                event="Notifications"
                noHeightAnimation
              >
                {open => (
                  <Observer>
                    {({ store }) => (
                      <Action
                        style={{ position: 'relative', fontSize: '1.25rem' }}
                        onClick={open}
                      >
                        <Tooltip
                          placement="bottom"
                          content={
                            store.userNotifications.unreadCount > 0
                              ? 'Show Notifications'
                              : 'No Notifications'
                          }
                        >
                          <BellIcon height={35}/>
                          {store.userNotifications.unreadCount > 0 && (
                            <UnreadIcon/>
                          )}
                        </Tooltip>
                      </Action>
                    )}
                  </Observer>
                )}
              </Overlay>
            )}

            <Action
              style={{ fontSize: '1.125rem' }}
              onClick={() =>
                modalOpened({
                  modal: 'newSandbox',
                })
              }
            >
              <Tooltip placement="bottom" content="New Sandbox">
                <PlusIcon height={35}/>
              </Tooltip>
            </Action>
          </Actions>

          {isLoggedIn ? <UserMenu/> : <SignInButton/>}
        </Wrapper>
      </Row>
    ),
  ),
);
