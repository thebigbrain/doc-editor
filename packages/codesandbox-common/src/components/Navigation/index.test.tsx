import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import Navigation from '.';

describe('<Navigation /> rendering', () => {
  it('basic', () => {
    const wrapper = mountWithTheme(<Navigation/>);
    expect(wrapper).toMatchSnapshot();
  });
});
