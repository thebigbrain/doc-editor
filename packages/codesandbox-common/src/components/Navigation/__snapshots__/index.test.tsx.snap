// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> rendering basic 1`] = `ReactWrapper {}`;
